lib/src/motorcontrol/sparkmax/SparkMaxPositionController.cpp

```diff
@@ -21,14 +21,17 @@ rmb::SparkMaxPositionController<U>::SparkMaxPositionController(int deviceID)
 template <typename U>
 rmb::SparkMaxPositionController<U>::SparkMaxPositionController(
     int deviceID, const PIDConfig &config, ConversionUnit_t conversionFactor,
-    const Feedforward<U> &ff, std::initializer_list<Follower> followerList)
+    const Feedforward<U> &ff, std::initializer_list<Follower> followerList, bool alternateEncoder)
     : sparkMax(deviceID, rev::CANSparkMax::MotorType::kBrushless),
       sparkMaxEncoder(sparkMax.GetEncoder()),
       sparkMaxPIDController(sparkMax.GetPIDController()),
       conversion(conversionFactor), feedforward(ff) {
 
   sparkMax.RestoreFactoryDefaults();
 
+  if(alternateEncoder)
+    (rev::RelativeEncoder&)sparkMaxEncoder = sparkMax.GetAlternateEncoder(1);
```

Comment:
This is disgusting code, but it seems to be necessary.

Reply:
That *is* gross, and it probably won’t work because the alternate encoder goes out of scope. You’ll likely need to make the encoder an ‘std::unique_ptr’ and then you’ll be able to use the terinary operator in the initializer list.